common/src/utils.py

```diff
@@ -0,0 +1,41 @@
+import os.path
+import sys
+from git import Repo
+
+# Path to root project git (if needed)
+ROOT_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), '../..'))
+PROJECT_ROOT_GIT_PATH = f'{ROOT_PATH}/.git'
+
+# Log messages to be printed by git_push
+COMMIT_MESSAGE = "Update shared repository\n"
+NO_CHANGES_MESSAGE = "No changes detected\n"
+DEBUGGING_MESSAGE = "Skipping git push because it's disabled for debugging.\n"
+PUSHING_MESSAGE = "Pushing files to shared repository\n"
+PUSH_SUCCESSFUL_MESSAGE = "Push successful\n"
+
+
+# Push shared repository to Git if any files changed
+def git_push(git_repo_path, log=sys.stdout, enable_push=True):
```

Comment:
Instead of declaring COMMIT_MESSAGE above, could you pass in commit_message as an argument here? Different callers might want different messages.

I would make the order of arguments:
def git_push(git_repo_path, commit_message="Update shared repository", enable_push=True, log=sys.stdout):
I see log as the least-likely to be used argument outside of testing, so putting it at the end

I would automatically add the "\n" onto the end of the commit message

Reply:
Done. Leaving off the "\n" entirely since part of the commit message, not being written to the log (the original "\n" wasn't needed for this message and was added in error).